### DISCUSSION TITLE
add CLI arg value if not present (--junitxml)

### DISCUSSION THREAD

**Discussion by choppsv1** [pytest-dev/pytest#8997] [Q&A] [answered]
I need to be able to add a dynamic default value for --junitxml, if it is not present based on other ini an cil args present. Currently I am using `--addopts --junitxml=/tmp/default.xml` in the an `pytest.ini` file.

I have another ini/cli option I add, `rundir`, which sets up where the run will happen and logs are placed, and so I'd like to dynamically set the junitxml, if unset to inside `rundir/default.xml`.

The place to do this seemed to be in my `pytest_configure()` hook in `conftest.py` where I process the `rundir` value. However I can't figure out how to add junit to the config at that point. Indeed is it too late to set junitxml in any case?

How can I manage to do this?

**Comment by nicoddemus** [comment, marked as answer]
Hi @choppsv1,

The internal JUnitXML plugin also checks the options in `pytest_configure`:

https://github.com/pytest-dev/pytest/blob/2439729413e701a19398316711ec5d795ae2ec63/src/_pytest/junitxml.py#L426-L427

Given that external plugins execute first, and then internal ones, your `pytest_configure()` should be able to set  `config.option.xmlpath` before `junitxml` plugin gets to it. Perhaps you are setting `config.option.junitxml` (after the command-line argument)?

**Reply by RonnyPfannschmidt** [reply]
i recall its possible with a `@pytest.hookimpl(tryfirst=True)` to set/force that value

**Reply by choppsv1** [reply]
> Hi @choppsv1,
> 
> The internal JUnitXML plugin also checks the options in `pytest_configure`:
> 
> https://github.com/pytest-dev/pytest/blob/2439729413e701a19398316711ec5d795ae2ec63/src/_pytest/junitxml.py#L426-L427
> 
> Given that external plugins execute first, and then internal ones, your `pytest_configure()` should be able to set `config.option.xmlpath` before `junitxml` plugin gets to it. Perhaps you are setting `config.option.junitxml` (after the command-line argument)?

This worked great, thanks!

```
    rundir = config.getini("rundir")
    if not rundir:
        rundir = "/tmp/topotests"
    if not config.getoption("--junitxml"):
        config.option.xmlpath = os.path.join(rundir, "topotests.xml")
```

**Reply by nicoddemus** [reply]
Cool. Can you mark my reply as "answer"? This helps others find it in the future. 👍
